### PR TITLE
MAINT: moving to PyAnsys actions

### PR DESCRIPTION
**Pull request by RobPasMue**
As title says. Also changing from poetry to flit =)

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -146,142 +83,80 @@ jobs:
...
   release:
     name: Release project
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
-    needs: [package, extras-import]
+    needs: [package, smoke-tests-extras, wheelhouse-all]
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: Release to the private PyPI repository
```
Comment (MaxJPRey) [inline .github/workflows/ci-build.yml:152]:
```suggestion
      - name: Release to PyPI repository
```

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -146,142 +83,80 @@ jobs:
...
-        run: |
-          pip install twine
-          twine upload --skip-existing ./**/*.whl
-          twine upload --skip-existing ./**/*.tar.gz
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          library-name: ${{ env.PACKAGE_NAME }}
+          twine-username: __token__
+          twine-token: ${{ secrets.PYANSYS_PYPI_PRIVATE_PAT }}
```
Comment (MaxJPRey) [inline .github/workflows/ci-build.yml:157]:
```suggestion
          twine-token: ${{ secrets.PYPI_TOKEN }}
```

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (MaxJPRey) [reply .github/workflows/ci-build.yml:152]: @RobPasMue . I think we want to keep publishing to public PyPI but I must have forgotten a point.

**Review by MaxJPRey — APPROVED** [review]
LGTM.

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (MaxJPRey) [reply .github/workflows/ci-build.yml:152]: If not, you might have to change the action used below.

Comment (RobPasMue) [reply .github/workflows/ci-build.yml:152]: Damn, you're totally right. I must have missed it.

**Pull request merged**